operator/src/lib.rs

```diff
@@ -235,36 +162,152 @@ impl OpaState {
                             warn!("No name found in metadata, this should not happen! Skipping node: [{:?}]", node);
                             continue;
                         };
-
-                        let pod_name = format!("{}-{}", name_prefix, node_name);
-
                         debug!(
-                            "Creating pod [{}] on node [{}] for [{}] role and group [{}]",
-                            pod_name,
+                            "Creating pod on node [{}] for [{}] role and group [{}]",
                             node.metadata
                                 .name
                                 .as_deref()
                                 .unwrap_or("<no node name found>"),
-                            node_type,
+                            role_str,
                             role_group
                         );
 
-                        // Create a pod for this node, role and group combination
-                        let pod = build_pod(
-                            &self.context.resource,
-                            node_name,
-                            &labels,
-                            &pod_name,
-                            &cm_name,
-                            opa_config,
-                        )?;
+                        let (pod, config_maps) = self
+                            .create_pod_and_config_maps(
+                                &role,
+                                role_group,
+                                &node_name,
+                                config_for_role_and_group(
+                                    role_str,
+                                    role_group,
+                                    &self.validated_role_config,
+                                )?,
+                            )
+                            .await?;
+
                         self.context.client.create(&pod).await?;
+
+                        for config_map in config_maps {
+                            self.create_config_map(config_map).await?;
+                        }
                     }
                 }
             }
         }
         Ok(ReconcileFunctionAction::Continue)
     }
+
+    async fn create_pod_and_config_maps(
```

Comment:
Could you add a comment on the logic in here?
Something that explains what's going on, how many configmaps there will be etc.
I know this code is obvious to you now but it's a lot of code without a single comment